scm-webapp/src/main/java/sonia/scm/api/v2/resources/MeDtoFactory.java

```diff
@@ -83,14 +83,16 @@ private PrincipalCollection getPrincipalCollection() {
 
   private MeDto createDto(User user) {
     Links.Builder linksBuilder = linkingTo().self(resourceLinks.me().self());
-    if (UserPermissions.delete(user).isPermitted()) {
-      linksBuilder.single(link("delete", resourceLinks.me().delete(user.getName())));
-    }
-    if (UserPermissions.modify(user).isPermitted()) {
-      linksBuilder.single(link("update", resourceLinks.me().update(user.getName())));
-    }
-    if (userManager.isTypeDefault(user) && UserPermissions.changePassword(user).isPermitted() && !Authentications.isSubjectAnonymous(user.getName())) {
-      linksBuilder.single(link("password", resourceLinks.me().passwordChange()));
+    if (isNotAnonymous(user)) {
```

Comment:
Why do we need this check? The anonymous user should not have those permissions.

Reply:
I have no idea how to prevent the anonymous user from get the permission on changing his own password.